src/platform/plugins/shared/chart_expressions/expression_partition_vis/public/utils/layers/get_color.ts

```diff
@@ -11,60 +11,42 @@ import { ArrayNode } from '@elastic/charts';
 import { isEqual } from 'lodash';
 import type { PaletteRegistry, SeriesLayer, PaletteOutput, PaletteDefinition } from '@kbn/coloring';
 import type { FieldFormatsStart } from '@kbn/field-formats-plugin/public';
-import type { FieldFormat } from '@kbn/field-formats-plugin/common';
 import { lightenColor } from '@kbn/charts-plugin/public';
-import type { Datatable } from '@kbn/expressions-plugin/public';
 import { BucketColumns, ChartTypes, PartitionVisParams } from '../../../common/types';
 import { DistinctSeries } from '../get_distinct_series';
-import { getNodeLabel } from './get_node_labels';
 
 const isTreemapOrMosaicChart = (shape: ChartTypes) =>
   [ChartTypes.MOSAIC, ChartTypes.TREEMAP].includes(shape);
 
 export const byDataColorPaletteMap = (
-  rows: Datatable['rows'],
-  column: Partial<BucketColumns>,
   paletteDefinition: PaletteDefinition,
   { params }: PaletteOutput,
-  formatters: Record<string, FieldFormat | undefined>,
-  formatter: FieldFormatsStart
+  colorIndexMap: Map<string, number>
 ) => {
-  const colorMap = new Map<string, string | undefined>(
-    rows.map((item) => {
-      const formattedName = getNodeLabel(
-        item[column.id ?? ''],
-        column,
-        formatters,
-        formatter.deserialize
-      );
-      return [formattedName, undefined];
-    })
-  );
-  let rankAtDepth = 0;
+  const colorCache = new Map<string, string | undefined>();
 
   return {
-    getColor: (item: unknown) => {
+    getColor: (item: string) => {
```

Comment:
This we will pass as an unformatted string. previously it was a formatted value but this is not necessary and breaks the `__other__` case since the categories are not formatted.